File: lab-2.2/c#/Human.cs
namespace CharacterLib
{
    public class Human
    {
        private string name;
        private int age;

        public Human()
        {
            name = "Unknown";
            age = 0;
        }

        public Human(string name, int age)
        {
            this.name = name;
            this.age = age;
        }

        public Human(Human other)
        {
            name = other.name;
            age = other.age;
        }

        public string GetName()
        {
            return name;
        }

        public int GetAge()
        {
            return age;
        }

        // Віртуальний метод
        public virtual string Identify()
        {
            return $"I am a human named {name}";
        }
    }
}

File: lab-2.2/c++/Human.cpp
#include "Human.h"

Human::Human() : name("Unknown"), age(0) {}

Human::Human(std::string n, int a) : name(n), age(a) {}

Human::Human(const Human& other) : name(other.name), age(other.age) {}

std::string Human::getName() const {
    return name;
}

int Human::getAge() const {
    return age;
}

std::string Human::identify() const {
    return "I am a human named " + name;
}
